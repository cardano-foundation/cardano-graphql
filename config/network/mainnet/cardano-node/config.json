{
  "AlonzoGenesisFile": "alonzo-genesis.json",
  "AlonzoGenesisHash": "1e696bd9cb57ee4baba7caf9685d90cf191092bb16b0118257694e4f1160ad45",
  "ByronGenesisFile": "byron-genesis.json",
  "ByronGenesisHash": "5f20df933584822601f9e3f8c024eb5eb252fe8cefb24d1317dc3d432e940ebb",
  "ConwayGenesisFile": "conway-genesis.json",
  "ConwayGenesisHash": "a567c908a7c25382e9d7adf25fa86212191d0a35fcd4ead5f8f40e52aa1faa7a",
  "EnableP2P": true,
  "LastKnownBlockVersion-Alt": 0,
  "LastKnownBlockVersion-Major": 3,
  "LastKnownBlockVersion-Minor": 0,
  "MaxKnownMajorProtocolVersion": 2,
  "Protocol": "Cardano",
  "RequiresNetworkMagic": "RequiresNoMagic",
  "ShelleyGenesisFile": "shelley-genesis.json",
  "ShelleyGenesisHash": "40710bcc4e024224641dc12fe1e5e478d2c3d3aa7ad4f032283ddcc73b2ddb03",
  "TargetNumberOfActivePeers": 20,
  "TargetNumberOfEstablishedPeers": 50,
  "TargetNumberOfKnownPeers": 100,
  "TargetNumberOfRootPeers": 100,
  "TraceAcceptPolicy": true,
  "TraceBlockFetchClient": false,
  "TraceBlockFetchDecisions": false,
  "TraceBlockFetchProtocol": false,
  "TraceBlockFetchProtocolSerialised": false,
  "TraceBlockFetchServer": false,
  "TraceChainDb": true,
  "TraceChainSyncBlockServer": false,
  "TraceChainSyncClient": false,
  "TraceChainSyncHeaderServer": false,
  "TraceChainSyncProtocol": false,
  "TraceConnectionManager": true,
  "TraceDNSResolver": true,
  "TraceDNSSubscription": true,
  "TraceDiffusionInitialization": true,
  "TraceErrorPolicy": true,
  "TraceForge": true,
  "TraceHandshake": true,
  "TraceInboundGovernor": true,
  "TraceIpSubscription": true,
  "TraceLedgerPeers": true,
  "TraceLocalChainSyncProtocol": false,
  "TraceLocalConnectionManager": true,
  "TraceLocalErrorPolicy": true,
  "TraceLocalHandshake": true,
  "TraceLocalRootPeers": true,
  "TraceLocalTxSubmissionProtocol": false,
  "TraceLocalTxSubmissionServer": false,
  "TraceMempool": false,
  "TraceMux": false,
  "TracePeerSelection": true,
  "TracePeerSelectionActions": true,
  "TracePublicRootPeers": true,
  "TraceServer": true,
  "TraceTxInbound": false,
  "TraceTxOutbound": false,
  "TraceTxSubmissionProtocol": false,
  "TracingVerbosity": "NormalVerbosity",
  "TurnOnLogMetrics": true,
  "TurnOnLogging": true,
  "defaultBackends": [
    "KatipBK"
  ],
  "defaultScribes": [
    [
      "StdoutSK",
      "stdout"
    ]
  ],
  "hasEKG": 12788,
  "hasPrometheus": [
    "127.0.0.1",
    12798
  ],
  "minSeverity": "Info",
  "options": {
    "mapBackends": {
      "cardano.node.metrics": [
        "EKGViewBK"
      ],
      "cardano.node.resources": [
        "EKGViewBK"
      ]
    },
    "mapSubtrace": {
      "cardano.node.metrics": {
        "subtrace": "Neutral"
      }
    }
  },
  "rotation": {
    "rpKeepFilesNum": 10,
    "rpLogLimitBytes": 5000000,
    "rpMaxAgeHours": 24
  },
  "setupBackends": [
    "KatipBK"
  ],
  "setupScribes": [
    {
      "scFormat": "ScText",
      "scKind": "StdoutSK",
      "scName": "stdout",
      "scRotation": null
    }
  ]
}