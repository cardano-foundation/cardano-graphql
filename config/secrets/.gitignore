postgres_db
postgres_user
postgres_password