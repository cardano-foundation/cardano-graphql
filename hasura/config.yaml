version: 2
endpoint: http://localhost:8090
metadata_directory: metadata
actions:
  kind: synchronous
  handler_webhook_baseurl: http://localhost:3000
