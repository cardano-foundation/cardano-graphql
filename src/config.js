import { multi } from 'getenv'

export default Object.assign({}, multi({
  SWAGGER_SCHEMA_URI: ['SWAGGER_SCHEMA_URI', undefined, 'url'],
  // Todo: Switch to 'url' type when swagger-to-graphql package properly resolves a trailing slash
  REST_ENDPOINT: ['REST_ENDPOINT', undefined, 'string'],
  PORT: ['PORT', 3000, 'string'],
  CONNECTION_ATTEMPT_INTERVAL: ['CONNECTION_ATTEMPT_INTERVAL', 1000, 'int'],
  MAX_CONNECTION_ATTEMPTS: ['MAX_CONNECTION_ATTEMPTS', 120, 'int'],
  TLS_PATH_CA: ['TLS_PATH_CA', '/data/auth/ca.cert', 'string'],
  TLS_PATH_KEY: ['TLS_PATH_KEY', '/data/auth/client.key', 'string'],
  TLS_PATH_CERT: ['TLS_PATH_CERT', '/data/auth/client.pem', 'string'],
  AUTH_DISABLED: ['AUTH_DISABLED', false, 'bool']
}), {
  get isProduction () { return process.env.NODE_ENV !== 'development' }
})
