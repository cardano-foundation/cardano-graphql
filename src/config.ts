import { Config as ServerConfig } from './Server'
import { MissingConfig } from './lib/errors'
import { buildHasuraSchema } from './lib/buildHasuraSchema'
import { hasuraResolvers, scalarResolvers } from './resolvers'

export async function getConfig (): Promise<ServerConfig> {
  const {
    apiPort,
    hasuraUri,
    queryDepthLimit,
    tracing
  } = filterAndTypecastEnvs(process.env)

  if (!hasuraUri) {
    throw new MissingConfig('HASURA_URI env not set')
  }

  return {
    apiPort: apiPort || 3100,
    context: hasuraUri ? await buildContext(hasuraUri) : undefined,
    queryDepthLimit: queryDepthLimit || 10,
    resolvers: Object.assign({}, scalarResolvers, hasuraResolvers),
    tracing
  }
}

function filterAndTypecastEnvs (env: any) {
  const {
    API_PORT,
    HASURA_URI,
    QUERY_DEPTH_LIMIT,
    TRACING
  } = env
  return {
    apiPort: Number(API_PORT),
    hasuraUri: HASURA_URI,
    queryDepthLimit: Number(QUERY_DEPTH_LIMIT),
    tracing: Boolean(TRACING)
  }
}

async function buildContext (hasuraUri: string) {
  const hasura = await buildHasuraSchema(hasuraUri)
  return () => ({ hasura })
}
