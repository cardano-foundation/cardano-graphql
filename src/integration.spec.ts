import * as path from 'path'
import * as fs from 'fs'
import gql from 'graphql-tag'
import { createTestClient, ApolloServerTestClient } from 'apollo-server-testing'
import { ApolloServerBase } from 'apollo-server-core'
import * as depthLimit from 'graphql-depth-limit'
import {
  block29021, block29022,
  epoch1,
  tx05ad8b, txe68043
} from './lib/data_assertions'
import { getConfig } from './config'

describe('Integration', () => {
  let apolloServer: ApolloServerBase
  let client: ApolloServerTestClient

  beforeEach(async () => {
    process.env.HASURA_URI = 'http://localhost:8090/v1/graphql'
    const { context, resolvers } = await getConfig()
    apolloServer = new ApolloServerBase({
      context,
      introspection: true,
      resolvers,
      typeDefs: fs.readFileSync(path.join(__dirname, 'schema.graphql'), 'UTF8'),
      validationRules: [depthLimit(20)]
    })
    client = createTestClient(apolloServer)
  }, 60000)

  describe('blocks', () => {
    it('returns the most recent block by default ', async () => {
      const result = (await client.query({
        query: gql`query {
            blocks {
                number
            }
        }`
      })).data.blocks

      const blockHeight = (await client.query({
        query: gql`query {
            cardano {
                blockHeight
            }
        }`
      })).data.cardano.blockHeight

      expect(result[0].number).toEqual(blockHeight)
      expect(result).toMatchSnapshot()
    })
    it('throws an error if query requests more than 100 blocks', async () => {
      const result = await client.query({
        query: gql`query {
            blocks (limit: 110) {
                id
            }
        }`
      })
      expect(result.errors[0]).toMatchSnapshot()
    })

    it('Uses pagination with an offset for larger result sets', async () => {
      const page1 = await client.query({
        query: gql`query {
            blocks (limit: 20, offset: 3, order_by: {number: asc}) {
                id
                number
            }
        }`
      })
      const page2 = await client.query({
        query: gql`query {
            blocks (limit: 20, offset: 23, order_by: {number: asc}) {
                id
                number
            }
        }`
      })
      expect(page1.data.blocks).toMatchSnapshot()
      expect(page2.data.blocks).toMatchSnapshot()
    })

    it('Can return blocks by number', async () => {
      const result = await client.query({
        query: gql`query {
            blocks (
                limit: 2,
                where: { number: { _eq: ${block29022.number}}}) {
                id
            }
        }`
      })
      expect(result.data.blocks.length).toBe(1)
      expect(result.data.blocks[0]).toEqual({ id: block29022.id })
      expect(result).toMatchSnapshot()
    })

    it('Can return blocks by an array of IDs', async () => {
      const result = await client.query({
        query: gql`query {
            blocks (
                limit: 2,
                where: { id: { _in: [
                  \"${block29021.id}\",
                  \"${block29022.id}\"
                ]}},
                order_by: { number: asc }
            ) {
                epoch {
                    number
                }
                fees
                id
                merkelRootHash
                number
                createdAt
                previousBlock {
                    id
                    number
                }
                size
                slotNo
                slotWithinEpoch
                transactions(order_by: {fee: desc}) {
                    block {
                        number
                    }
                    fee
                    id
                    includedAt
                    inputs {
                        address
                        value
                    }

                    outputs {
                        value
                        address
                    }
                    totalOutput
                }
            }
        }`
      })
      expect(result.data.blocks.length).toBe(2)
      expect(result.data.blocks[0]).toEqual(block29021)
      expect(result.data.blocks[1]).toEqual(block29022)
      // expect(result.data.blocks[1]).toEqual(block43178)
      // expect(result).toMatchSnapshot()
    })
    it('are linked to their predecessor, and the chain can be traversed', async () => {
      const result = await client.query({
        query: gql`query {
            blocks (where: { number: { _eq: ${block29022.number}}}) {
                id
                previousBlock {
                    number
                    previousBlock {
                        number
                        previousBlock {
                            number
                        }
                    }
                }
            }
        }`
      })
      expect(result.data.blocks[0].previousBlock.previousBlock.previousBlock.number).toBe(29019)
      expect(result).toMatchSnapshot()
    })
  })

  describe('epochs', () => {
    const noBoundErrorMessage = 'number must be specified (_eq) or bounded (_in)'
    const boundTooLargeMessage = 'Maximum number of epochs queryable in a range is 10'

    it('throws if number is not specified in the where clause', async () => {
      const result = await client.query({
        query: gql`query {
            epochs {
                output
                number
                transactionsCount
            }
        }`
      })

      expect(result.errors[0].message).toEqual(noBoundErrorMessage)
    })

    it('throws if the bound specified in the where clause is too large', async () => {
      const result = await client.query({
        query: gql`query {
            epochs(where: {number: {_in: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]}}) {
                output
                number
                transactionsCount
            }
        }`
      })

      expect(result.errors[0].message).toEqual(boundTooLargeMessage)
    })

    it('Returns epoch details by number', async () => {
      const result = await client.query({
        query: gql`query {
            epochs( where: { number: { _eq: ${epoch1.number} }}) {
                output
                number
                transactionsCount
                startedAt
                lastBlockTime
            }
        }`
      })
      expect(result.data.epochs[0]).toEqual(epoch1)
      expect(result).toMatchSnapshot()
    })

    it('Returns epoch details by number range', async () => {
      const result = await client.query({
        query: gql`query {
            epochs( where: { number: { _in: [${epoch1.number}] }}) {
                output
                number
                transactionsCount
                startedAt
                lastBlockTime
            }
        }`
      })
      expect(result.data.epochs[0]).toEqual(epoch1)
      expect(result).toMatchSnapshot()
    })

    it('Returns blocks scoped to epoch', async () => {
      const validQueryResult = await client.query({
        query: gql`query {
            epochs( where: { number: { _eq: 1 }}) {
                blocks(limit: 20) {
                    id
                }
            }
        }`
      })
      const invalidQueryResult = await client.query({
        query: gql`query {
            epochs( where: { number: { _eq: 1 }}) {
                blocks(limit: 20, where: { epoch: { number: { _eq: 0 } }}) {
                    id
                }
            }
        }`
      })
      expect(validQueryResult.data.epochs[0].blocks.length).toBe(20)
      expect(invalidQueryResult.data.epochs[0].blocks.length).toBe(0)
    })
  })

  describe('cardano', () => {
    it('Returns the block height', async () => {
      const result = await client.query({
        query: gql`query {
            cardano {
                blockHeight
                currentEpoch {
                    number
                }
            }
        }`
      })
      expect(result.data.cardano.blockHeight).toBe(31070)
      expect(result.data.cardano.blockHeight).toMatchSnapshot()
    })
  })

  // describe('stakePools', () => {
  //   it('Returns Stake Pools', async () => {
  //     const result = await client.query({
  //       query: gql`query {
  //           stakePools {
  //               controlledStake
  //               id
  //               ticker
  //           }
  //       }`
  //     })
  //     expect(result).toMatchSnapshot()
  //   })
  // })

  describe('transactions', () => {
    it('Returns transactions by IDs', async () => {
      const result = await client.query({
        query: gql`query {
            transactions(
                limit: 2,
                where: { id: { _in: [\"${txe68043.id}\", \"${tx05ad8b.id}\"]}},
                order_by: {fee: desc}
            ) {
                block {
                    number
                }
                fee
                id
                inputs {
                    address
                    value
                }
                outputs {
                    address
                    value
                }
                totalOutput
            }
        }`
      })
      expect(result.data.transactions.length).toBe(2)
      expect(result).toMatchSnapshot()
    })

    describe('utxoSet', () => {
      it('Returns the whole set by default', async () => {
        const result = await client.query({
          query: gql`query {
              utxoSet(
                  limit: 20
              ) {
                  address
                  value
              }
          }`
        })
        expect(result.data.utxoSet.length).toBe(20)
        expect(result).toMatchSnapshot()
      })
      it('Can be filtered by address', async () => {
        const result = await client.query({
          query: gql`query {
              utxoSet(
                  where: { address: { _eq:
                  "DdzFFzCqrhskotfhVwhLvNFaVGpA6C4yR9DXe56oEL4Ewmze51f1uQsc1cQb8qUyqgzjUPBgFZiVbuQu7BaXrQkouyvzjYjLqfJpKG5s"
                  }
                  }
              ) {
                  address
                  value
              }
          }`
        })
        expect(result.data.utxoSet.length).toBe(1)
        expect(result).toMatchSnapshot()
      })
    })
  })
})
