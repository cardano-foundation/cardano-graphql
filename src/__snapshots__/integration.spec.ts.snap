// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Integration blocks Can return blocks by number 1`] = `
Object {
  "data": Object {
    "blocks": Array [
      Object {
        "id": "bf13d9a80ad99a4f34edb8a3262dd8120e29bbe182732cd3b00bf3d1bb7c2380",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Integration blocks Uses pagination with an offset for larger result sets 1`] = `
Array [
  Object {
    "id": "be06c81f4ad34d98578b67840d8e65b2aeb148469b290f6b5235e41b75d38572",
    "number": 4,
  },
  Object {
    "id": "46debe49b4fe0bc8c07cfe650de89632ca1ab5d58f04f8c88d8102da7ef79b7f",
    "number": 5,
  },
  Object {
    "id": "365201e928da50760fce4bdad09a7338ba43a43aff1c0e8d3ec458388c932ec8",
    "number": 6,
  },
  Object {
    "id": "e39d988dd815fc2cb234c2abef0d7f57765eeffb67331814bdb01c590359325e",
    "number": 7,
  },
  Object {
    "id": "2d9136c363c69ad07e1a918de2ff5aeeba4361e33b9c2597511874f211ca26e9",
    "number": 8,
  },
  Object {
    "id": "f4b2213e2ed6dc8d4da0f4234ced91714c8cbaa0f3750abb90d10aff23ce4d67",
    "number": 9,
  },
  Object {
    "id": "9cf9488161d63c7c414008737e1fab2eb8b85bf55dfe3fd8a09d2bc0c4327fe3",
    "number": 10,
  },
  Object {
    "id": "4bcd56b2e530a9690a94fb12a03cfd2cb7f25cdb229fda545c8cfddb57e13e24",
    "number": 11,
  },
  Object {
    "id": "7054258f4310b7803419681db10cd4a47707c5d5182ec9317fae7e2e93a399af",
    "number": 12,
  },
  Object {
    "id": "8b72b27a29fbe95f925f97afa0ce81c2e9aface13a539c733733153dadb8c928",
    "number": 13,
  },
  Object {
    "id": "5701cdc2396dc3f3ee32a13ea538d9a24059c3d00e7b928da3172835504001a9",
    "number": 14,
  },
  Object {
    "id": "8ee800d2d2cb242a5fe96a13371118a3ceb1cd068d5daa91385c351954bece35",
    "number": 15,
  },
  Object {
    "id": "d7900f547f96465c55ea3d8e6a9f6aeb88be200ad2138540eee716e94c1d211c",
    "number": 16,
  },
  Object {
    "id": "d0d37e4c9eabfe893c9c2cdaab7e90d450e570462a4dd315b6e7a6dcb54bd29a",
    "number": 17,
  },
  Object {
    "id": "0be303663930359a7c2d2d039458cd537110863c24f475bffb09b19dc3f9d54a",
    "number": 18,
  },
  Object {
    "id": "29dd9773213662145e5e399e3bc5dd2e86ef1f563bfb5309c552dc4349632278",
    "number": 19,
  },
  Object {
    "id": "7c6901c6346781c2bc5cbc49577490e336c2545c320ce4a61605bc71a9c5bed0",
    "number": 20,
  },
  Object {
    "id": "336113931d88269b5b56e7988fb4f9469e69aaa67ff84c6eef0f3e5adf1af442",
    "number": 21,
  },
  Object {
    "id": "ae230b33767ee2f0492347f062a48976cfd4367987c1a96fa5de52eba3268f5f",
    "number": 22,
  },
  Object {
    "id": "a076c4b645ec23e7e6fa909821edb3e28106d0ffa6325cd2d32e0f3b53c1c09b",
    "number": 23,
  },
]
`;

exports[`Integration blocks Uses pagination with an offset for larger result sets 2`] = `
Array [
  Object {
    "id": "99627f41664532199a53f90686e2370d4b83b1b7ed36d49f91c4b87ffd119b94",
    "number": 24,
  },
  Object {
    "id": "0f1f41d1d29a7f4e601ab00deb07eeaf7a56c14cf554f5189d88f22157bb2844",
    "number": 25,
  },
  Object {
    "id": "b6e5f58337dfb37da5f0ceedb127b808db72c14f86711fe69fadeadce54d17f1",
    "number": 26,
  },
  Object {
    "id": "30830e8bd51959ddee37e5a3517d02d52d5f9b9c416edb7571a3e90e3a588599",
    "number": 27,
  },
  Object {
    "id": "3265de2d67fd906f0c1316a10a8286656f749fa05f3fa0f26e6216a859d349fc",
    "number": 28,
  },
  Object {
    "id": "35837ea76e42d5f0bc3309127c9deb92e87d449ffc7d864d5eee3cbd63b2574a",
    "number": 29,
  },
  Object {
    "id": "d3fdc8c8ea4050cc87a21cb73110d54e3ec92f8ae76941e8a1957ed6e6a7e0b0",
    "number": 30,
  },
  Object {
    "id": "bfd2d6b38a6e16e5710d5e6eedefaf30fe7e15753a4cb9e8da95d0a9b10525c5",
    "number": 31,
  },
  Object {
    "id": "cd7c2415cffa1de98f87fadad6329045c4342637336f1c8b2e016f25f3c7fac5",
    "number": 32,
  },
  Object {
    "id": "1009f9569b4edaf2fc47803f7c623418f7e2d46d561cd85995edf50533a0c581",
    "number": 33,
  },
  Object {
    "id": "26b99991a7220932651eea69733dfda48dd187f3d0b44517ca334067958372b4",
    "number": 34,
  },
  Object {
    "id": "151d7119e85f5bc9400fea907731a1cf2e8800b7a2828d1a068902aaebcf2081",
    "number": 35,
  },
  Object {
    "id": "f4ee7dd414985862e03862688a3acbdb06bef42c09042a9b9ba4d63cc675a311",
    "number": 36,
  },
  Object {
    "id": "7f9f47d868f688255a0b7a04b7337ad8827edc37bedf8ee21166e1c989e2ef0e",
    "number": 37,
  },
  Object {
    "id": "1be4840723bc6c6ee2c3a23ba64c216a478a9654f360382ed31231af6593fe1d",
    "number": 38,
  },
  Object {
    "id": "61032e54e26679b8384caf62df8024665da80f0dc81ae5b76152c9d55a705fdb",
    "number": 39,
  },
  Object {
    "id": "8bf5c27d07e57fb5e843a5d9088928180afbf8c733929b6a84fc95c5df670e5d",
    "number": 40,
  },
  Object {
    "id": "1dcac253454fe43882bb4921b2a305a332adbdacca5aafb66829351035058277",
    "number": 41,
  },
  Object {
    "id": "8675f84d2d0dfe12cefeb6bfe2888fbb80a0b753fe4dbf612edfb70151523192",
    "number": 42,
  },
  Object {
    "id": "2a5ad3f0333b8118ec17791ca834d2d8760e853f3d319f4f00fb805bb2953cb0",
    "number": 43,
  },
]
`;

exports[`Integration blocks are linked to their predecessor, and the chain can be traversed 1`] = `
Object {
  "data": Object {
    "blocks": Array [
      Object {
        "id": "bf13d9a80ad99a4f34edb8a3262dd8120e29bbe182732cd3b00bf3d1bb7c2380",
        "previousBlock": Object {
          "number": 29021,
          "previousBlock": Object {
            "number": 29020,
            "previousBlock": Object {
              "number": 29019,
            },
          },
        },
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Integration blocks returns the most recent block by default  1`] = `
Array [
  Object {
    "number": 31070,
  },
]
`;

exports[`Integration blocks throws an error if query requests more than 100 blocks 1`] = `[GraphQLError: 110 exceeds the maximum allowed value of 100. Use the offset to paginate through a larger result set]`;

exports[`Integration cardano Returns the block height 1`] = `31070`;

exports[`Integration epochs Returns epoch details by number 1`] = `
Object {
  "data": Object {
    "epochs": Array [
      Object {
        "endedAt": null,
        "number": 1,
        "output": "17282903106017760",
        "startedAt": "2017-09-28T21:45:51",
        "transactionsCount": "5344",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Integration epochs Returns epoch details by number range 1`] = `
Object {
  "data": Object {
    "epochs": Array [
      Object {
        "endedAt": null,
        "number": 1,
        "output": "17282903106017760",
        "startedAt": "2017-09-28T21:45:51",
        "transactionsCount": "5344",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Integration transactions Returns transactions by IDs 1`] = `
Object {
  "data": Object {
    "transactions": Array [
      Object {
        "block": Object {
          "number": 29021,
        },
        "fee": "171246",
        "id": "05ad8b467095e0886713a38231ab9fe84e4031a433a9400ebf70ec9415e20102",
        "inputs": Array [
          Object {
            "address": "DdzFFzCqrht5ExAdoZVExXoZTpoMYGKxva3thvvvHapsLZQzSX3kCqwgqi5NSM2oUtHYYDqsSnvSGbqkarB6cSgDZohUhLWZ9KFdDWsa",
            "value": "4924799649906",
          },
        ],
        "outputs": Array [
          Object {
            "address": "DdzFFzCqrhsxVv8Sjuj35Zkw42T9WjwXa6DasrNknm5M3bPVDnWJWwhX5NymoZncUArTfFYZ9sYGaA9htKSNYoK5Vkzx2vUtunAFt3eE",
            "value": "4824799478660",
          },
          Object {
            "address": "DdzFFzCqrhskotfhVwhLvNFaVGpA6C4yR9DXe56oEL4Ewmze51f1uQsc1cQb8qUyqgzjUPBgFZiVbuQu7BaXrQkouyvzjYjLqfJpKG5s",
            "value": "100000000000",
          },
        ],
        "totalOutput": "4924799478660",
      },
      Object {
        "block": Object {
          "number": 29021,
        },
        "fee": "171070",
        "id": "e680432562b3b71fe44ca4eb8e29cb181d3a0858b3e2a643a55f7513d901bcae",
        "inputs": Array [
          Object {
            "address": "DdzFFzCqrhsggyaAMAUTjGtjBr1CTp8tTcHYWbqtoyQBZcaYHM16rjbUDawTwoVaEPawAMPLJmpJVXHBNxZnTgmQzqAcNDe6XvMe5BkB",
            "value": "768403000000",
          },
        ],
        "outputs": Array [
          Object {
            "address": "DdzFFzCqrhsuz652nVpjktdtiV44uWJLHv83m61S33gzfB4TBx7SKp3DgM18fBJznMrbUdsEFEvXW4LYqVKFE9fjMgVhmJP2LBhUvEe8",
            "value": "768401828930",
          },
          Object {
            "address": "DdzFFzCqrhsgrGt7zHydkE4cKxjB7SgsdnMPUFxYEwBu1u4sGobxZL5Xo3eDycCrtSYAUyw9aAsZZxFpTdgur6MwYDYMwAZ4xNGKRD3o",
            "value": "1000000",
          },
        ],
        "totalOutput": "768402828930",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Integration transactions utxoSet Can be filtered by address 1`] = `
Object {
  "data": Object {
    "utxoSet": Array [
      Object {
        "address": "DdzFFzCqrhskotfhVwhLvNFaVGpA6C4yR9DXe56oEL4Ewmze51f1uQsc1cQb8qUyqgzjUPBgFZiVbuQu7BaXrQkouyvzjYjLqfJpKG5s",
        "value": "100000000000",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Integration transactions utxoSet Returns the whole set by default 1`] = `
Object {
  "data": Object {
    "utxoSet": Array [
      Object {
        "address": "Ae2tdPwUPEZ9vtyppa1FdJzvqJZkEcXgdHxVYAzTWcPaoNycVq5rc36LC1S",
        "value": "538861000000",
      },
      Object {
        "address": "Ae2tdPwUPEZGvXJ3ebp4LDgBhbxekAH2oKZgfahKq896fehv8oCJxmGJgLt",
        "value": "1153846000000",
      },
      Object {
        "address": "Ae2tdPwUPEYyahAbgfTxhChyWV6xMWKCDQs35Cr9qUXgzVviBrvhJ6NyUzA",
        "value": "455585000000",
      },
      Object {
        "address": "Ae2tdPwUPEZLC5VMKspod9dcf5PtUtvobxd3THtoxfR9uuzcqDghef9oiTc",
        "value": "390975000000",
      },
      Object {
        "address": "Ae2tdPwUPEZ415vhqz7i5t8jP5dKsDHUmTpTGf4XxeezzJJzhaubdWm4Aj2",
        "value": "378507000000",
      },
      Object {
        "address": "Ae2tdPwUPEZ5GNoihaqR2Q43dTxBu9e9Y8m3daWcVw1bvswn4Y8APVUHqaZ",
        "value": "499999000000",
      },
      Object {
        "address": "Ae2tdPwUPEZH6yx1bPBeTVy2ikCuJ2aLmsXpYT622bTw6LEy9UHiu9k8KF5",
        "value": "411085000000",
      },
      Object {
        "address": "Ae2tdPwUPEZDjmVmgpcP1Dcm2C7frvbdZ8XvJjAPNVxLodknP4XACUTiidE",
        "value": "901105000000",
      },
      Object {
        "address": "Ae2tdPwUPEYyeBiwUSF6e1zXBY2Rbc8jEpXgcueqUyvkTaJekBw3sTbeMBG",
        "value": "500919000000",
      },
      Object {
        "address": "Ae2tdPwUPEZ9BzUH1zgCijhBep4ZnwUhmM2zS7nkmnZV7gvbV54f5oCxxX7",
        "value": "462355000000",
      },
      Object {
        "address": "Ae2tdPwUPEZKsHEnJoZZkUccHv1vsztJ37amLwxxdJpFYicuLwRpKFgsEjn",
        "value": "391257000000",
      },
      Object {
        "address": "Ae2tdPwUPEZ2cE1drgm8LKf7L6QVqEmov7eYegL6S7W3L7ZHu2q3AxiU6PW",
        "value": "7692307000000",
      },
      Object {
        "address": "Ae2tdPwUPEZKxVVgakuw9iytSjfYzEbDhjVCwmWQpgyuNcPxtueMujWDAEu",
        "value": "469217000000",
      },
      Object {
        "address": "Ae2tdPwUPEYwEuhFnowz2h8rxJ9JoNbjiU6H7dL1ed5RLMikDr6qUJsJ9jt",
        "value": "383233000000",
      },
      Object {
        "address": "Ae2tdPwUPEZ3qXMX6imKEXTGPVBtY1QAx1oroD5Mf4buQiSvaJqi3qXVA3R",
        "value": "1000000",
      },
      Object {
        "address": "Ae2tdPwUPEZBA28BxMDYxGFHGS3MWaa6ioT9ZBnCoy8ojtPy6YcapAvXhGq",
        "value": "416901000000",
      },
      Object {
        "address": "Ae2tdPwUPEZBZTsRj7nGvvWQDTkqD9KLpCPpuZvjA1roL7KLDDVgkPU5S8j",
        "value": "385506000000",
      },
      Object {
        "address": "Ae2tdPwUPEZAVCQEGtLWUy5JhzoJCqAaaN9yvAZAGgjyZT49SjLnzsZeMzX",
        "value": "500305000000",
      },
      Object {
        "address": "Ae2tdPwUPEZA6N6PzxtcoP4PKrBQBLM4FTB89q4zaWnEBdbCD8QTo51zPCn",
        "value": "380005000000",
      },
      Object {
        "address": "Ae2tdPwUPEZH6K1hZZTC7T1zcZpKQdWmxUdGQtf8bix78EULKVnDQhQYYY3",
        "value": "377802000000",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
